src/test/java/org/folio/edge/rtac/RtacHandlerTest.java

```diff
@@ -0,0 +1,145 @@
+package org.folio.edge.rtac;
+
+import static org.apache.http.HttpStatus.SC_NOT_ACCEPTABLE;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.folio.edge.core.Constants.APPLICATION_JSON;
+import static org.folio.edge.core.Constants.APPLICATION_XML;
+import static org.folio.edge.core.Constants.TEXT_PLAIN;
+import static org.junit.Assert.assertEquals;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.restassured.RestAssured;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.edge.rtac.model.Instances;
+import org.folio.edge.rtac.utils.RtacMockOkapi;
+import org.junit.Test;
+
+public class RtacHandlerTest extends MainVerticleTest {
+
+  private static final Logger logger = LogManager.getLogger(RtacHandlerTest.class);
+
+  private String prepareQueryFor(String apiKey, String... instanceIds) {
+    if (ArrayUtils.isEmpty(instanceIds)) {
+      return String.format("/rtac?apikey=%s", apiKey);
+    } else {
+      String instancesAsString = Arrays.asList(instanceIds).stream()
+          .collect(Collectors.joining(","));
+      return String.format("/rtac?apikey=%s&instanceIds=%s", apiKey, instancesAsString);
+    }
+  }
+
+  private Instances prepareRecordsFor(String... instanceIds) {
+    if (ArrayUtils.isEmpty(instanceIds)) {
+      throw new IllegalArgumentException("No instances specified");
+
+    } else {
+      final var holdings = Arrays.asList(instanceIds).stream().map(RtacMockOkapi::getHoldings)
+          .collect(Collectors.toList());
+      final var instanceHoldingRecords = new Instances();
+      instanceHoldingRecords.setHoldings(holdings);
+      return instanceHoldingRecords;
+    }
+  }
+
+  @Test
+  public void checkNoAcceptHeadersSpecifiedSuccessfulResponse() throws IOException {
```

Comment:
I think test names can provide a useful summary of the intent of a test, I might be tempted to rename this to something like `shouldRespondWithXMLWhenClientDoesNotStateAPreference`.

```suggestion
  public void shouldRespondWithXMLWhenClientDoesNotStateAPreference() throws IOException {
```

Reply:
Yes, that was good consideration from semantic point of view, I'll fix it. Thank you.